src/yarp.c

```diff
@@ -6129,7 +6129,9 @@ parse_expression_prefix(yp_parser_t *parser, binding_power_t binding_power) {
 
       yp_undef_node_append(undef, name);
 
-      while (accept(parser, YP_TOKEN_COMMA)) {
+      while (match_type_p(parser, YP_TOKEN_COMMA)) {
+        lex_state_set(parser, YP_LEX_STATE_FNAME | YP_LEX_STATE_FITEM);
```

Comment:
This matches [this line](https://github.com/ruby/ruby/blob/f313514563853f1ee623135c2959df2b518ee755/parse.y#L2567) in Ripper's `parse.y`